utils/EventRepository.js

```diff
@@ -225,15 +81,9 @@ class EventRepository {
       );
     }
 
-    // modify the url for event specific endpoints
-    const url = this.url.slice(-1) == "/" ? this.url.slice(0, -1) : this.url;
-    // add items to the queue
-    this.payloadQueue.addItem({
-      url: `${url}/v1/${type}`,
-      headers,
-      message,
-    });
+    this.queue.enqueue(headers, message, type);
```

Comment:
I don't think the headers defined in Ln67 are applicable for Beacon.

Reply:
You can move the headers preparation responsibility to the queues.

Reply:
Used in BeaconQueue.js --> fn flushQueue --> const blob = new Blob([payload], headers);

Reply:
Beacon queue ignores the headers I believe, which is why the write key is being passed as a query parameter. Can you please verify that it is part of the original implementation?

Reply:
Yes it is part of the original implementation. But not needed. new Blob([payload], headers) - only takes the data and the type. Which in this case is application/json.